test/test_prototype_builtin_datasets.py

```diff
@@ -169,21 +155,14 @@ def test_infinite_buffer_size(self, test_home, dataset_mock, config):
         dataset = datasets.load(dataset_mock.name, **config)
 
         for dp in extract_datapipes(dataset):
-            if isinstance(
-                dp,
-                (
-                    Shuffler,
-                    Demultiplexer,
-                    Forker,
-                    Grouper,
-                    MaxTokenBucketizer,
-                    UnZipper,
-                    IterKeyZipper,
-                ),
-            ):
-                # TODO: replace this with the proper sentinel as soon as https://github.com/pytorch/data/issues/335 is
-                #  resolved
-                assert dp.buffer_size == INFINITE_BUFFER_SIZE
+            try:
+                buffer_size = getattr(dp, "buffer_size")
+            except AttributeError:
+                continue
```

Comment:
why not `if hasattr`?

Reply:
> The first snippet you wrote is also an early return right?

Yeah, but I prefer the other style.

> I won't fight for this but the code we have right now is literally re-implementing the builtin [hasattr](https://docs.python.org/3/library/functions.html#hasattr)

I agree, my solution was an overkill. This is why I already fixed it in 7e36fd9.